processing/src/main/java/io/druid/segment/filter/Filters.java

```diff
@@ -208,73 +214,167 @@ public static ImmutableBitmap matchPredicate(
+    return selector.getBitmapFactory()
+                   .union(makePredicateQualifyingBitmapIterable(bitmapIndex, predicate, dimValues));
+  }
+
+  /**
+   * Return an estimated selectivity for bitmaps of all values matching the given predicate.
+   *
+   * @param columnSelector column selector
+   * @param dimension      dimension to look at
+   * @param indexSelector  bitmap selector
+   * @param predicate      predicate to use
+   *
+   * @return estimated selectivity
+   *
+   * @see Filters#matchPredicate(String, BitmapIndexSelector, Predicate)
+   */
+  static double estimatePredicateSelectivity(
+      final ColumnSelector columnSelector,
+      final String dimension,
+      final BitmapIndexSelector indexSelector,
+      final Predicate<String> predicate
+  )
+  {
+    Preconditions.checkNotNull(dimension, "dimension");
+    Preconditions.checkNotNull(indexSelector, "selector");
+    Preconditions.checkNotNull(predicate, "predicate");
+
+    // Missing dimension -> match all rows if the predicate matches null; match no rows otherwise
+    final Indexed<String> dimValues = indexSelector.getDimensionValues(dimension);
+    if (dimValues == null || dimValues.size() == 0) {
+      return predicate.apply(null) ? 1. : 0.;
+    }
+
+    // Apply predicate to all dimension values and union the matching bitmaps
+    final BitmapIndex bitmapIndex = indexSelector.getBitmapIndex(dimension);
+    return estimatePredicateSelectivity(
+        columnSelector,
+        dimension,
+        makePredicateQualifyingBitmapIterable(
+            bitmapIndex,
+            predicate,
+            dimValues
+        ),
+        indexSelector.getNumRows()
+    );
+  }
+
+  static double estimatePredicateSelectivity(
+      ColumnSelector columnSelector,
+      String dimension,
+      Iterable<ImmutableBitmap> bitmaps,
+      long totalNumRows
+  )
+  {
+    final ColumnCapabilities columnCapabilities = columnSelector.getColumn(dimension).getCapabilities();
+    return estimatePredicateSelectivity(
+        bitmaps,
+        totalNumRows,
+        columnCapabilities == null || columnCapabilities.hasMultipleValues()
+    );
+  }
+
+  static double estimatePredicateSelectivity(
+      Iterable<ImmutableBitmap> bitmaps,
+      long totalNumRows,
+      boolean isMultiValueDimension
+  )
+  {
+    long numMatchedRows = 0;
+    final List<ImmutableBitmap> bitmapList = Lists.newArrayList(bitmaps);
+    for (ImmutableBitmap bitmap : bitmapList) {
+      numMatchedRows += bitmap.size();
+    }
+
+    // assume multi-value column if columnCapabilities is null
+    if (isMultiValueDimension) {
+      final double estimated = numMatchedRows * Filters.computeNonOverlapRatioFromBitmapSamples(bitmapList)
+                               / totalNumRows;
+      return Math.min(1., estimated);
+    } else {
+      return (double) numMatchedRows / totalNumRows;
+    }
+  }
+
+  private static double computeNonOverlapRatioFromBitmapSamples(List<ImmutableBitmap> bitmaps)
```

Comment:
I don't understand this method. Could you please add an explanation? Ideally add some tests.